sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,164 @@
+        cleanRequest.httpBody = regexSubscriptionReplace(String(data: request.httpBody, encoding: .utf8))?.data(using: .utf8)
+    
+        return cleanRequest
+    }
+    
+    static func regexSubscriptionReplace(_ string: String?)->String?{
+        guard string != nil else {
+            return nil
+        }
+        var editedString = string!
+        var range = NSRange(location: 0, length: editedString.utf8.count)
+        if subscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = subscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        range = NSRange(location: 0, length: editedString.utf8.count)
+        if graphSubscriptionIDRegex.numberOfMatches(in: editedString, options: [], range: range) > 0 {
+            editedString = graphSubscriptionIDRegex.stringByReplacingMatches(in: editedString, options: [], range: range, withTemplate: "$1/99999999-9999-9999-9999-999999999999")
+        }
+        return editedString
+    }
+    
+    static func scrubSubscriptionIDs(response: Foundation.URLResponse, data: Data?) -> URLRequest {
+        fatalError("Unimplemented")
+    }
+    
+    public static func scrubRequests(request: URLRequest) -> URLRequest {
+        var cleanedRequest = request
+        cleanedRequest = scrubSubscriptionIDs(request: cleanedRequest)
+        
+        return cleanedRequest
+    }
+    
+    static func scrubResponse(response: Foundation.URLResponse,data: Data?) -> (Foundation.URLResponse, Data?) {
+        fatalError("Unimplemented")
+    }
+}
+
+
+extension Array where Element == Any {
+    func dictionaryForIndex(_ index: Int) throws -> Dictionary<String,Any> {
+        if let dictionary = self[index] as? Dictionary<String, Any> {
+            return dictionary
+        }
+        throw "Not a dictionary"
+    }
+    
+    func arrayForIndex(_ index: Int) throws -> Array<Any> {
+        if let array = self[index] as? Array<Any> {
+            return array
+        }
+        throw "Not an array"
+    }
+    
+    func stringForIndex(_ index: Int) throws -> String {
+        if let string = self[index] as? String {
+            return string
+        }
+        throw "Not a string"
+    }
+    
+}
+
+ extension Dictionary where Key == String, Value == Any {
+    
+    static func findValue(key: String, dictionary: Dictionary<String,Any>) -> Any? {
+       if dictionary[key] != nil {
+           return dictionary[key]
+       }
+        
+       for testKey in dictionary.keys {
+           if let innerDictionary = dictionary[testKey] as? Dictionary<String , Any> {
+               let returnedValue = findValue(key: key, dictionary: innerDictionary)
+               if returnedValue != nil {
+                   return returnedValue
+               }
+           }
+       }
```

Comment:
It looks like you are recursively looking for a key and stop once the first match is found. This seems like it may be error-prone. For example, if both request and response have a uri field, and you search for uri, you'll get back only one, and the other may not be reachable at all with this logic.